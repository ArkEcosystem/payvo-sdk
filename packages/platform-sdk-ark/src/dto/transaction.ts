import { Contracts, DTO } from "@arkecosystem/platform-sdk";
import { DateTime } from "@arkecosystem/platform-sdk-intl";
import { BigNumber } from "@arkecosystem/platform-sdk-support";

export class TransactionData extends DTO.AbstractTransactionData implements Contracts.TransactionData {
	public id(): string {
		return this.data.id;
	}

	public blockId(): string | undefined {
		return this.data.blockId;
	}

	public timestamp(): DateTime | undefined {
		return DateTime.fromUnix(this.data.timestamp.unix);
	}

	public confirmations(): BigNumber {
		return BigNumber.make(this.data.confirmations);
	}

	public sender(): string {
		return this.data.sender;
	}

	public recipient(): string {
		return this.data.recipient;
	}

	public recipients(): Contracts.MultiPaymentRecipient[] {
		if (!this.isMultiPayment()) {
			return [];
		}

		return this.data.asset.payments.map((payment: { recipientId: string; amount: string }) => ({
			address: payment.recipientId,
			amount: BigNumber.make(payment.amount),
		}));
	}

	public amount(): BigNumber {
		if (this.isMultiPayment()) {
			return this.data.asset.payments.reduce(
				(sum: BigNumber, { amount }: { amount: string }) => sum.plus(amount),
				BigNumber.ZERO,
			);
		}

		return BigNumber.make(this.data.amount);
	}

	public fee(): BigNumber {
		return BigNumber.make(this.data.fee);
	}

	public memo(): string | undefined {
		return this.data.vendorField;
	}

	public asset(): Record<string, unknown> {
		return this.data.asset;
	}

	public isConfirmed(): boolean {
		return this.confirmations().isGreaterThanOrEqualTo(51);
	}

	public isSent(): boolean {
		return [this.getMeta("address"), this.getMeta("publicKey")].includes(this.sender());
	}

	public isReceived(): boolean {
		return [this.getMeta("address"), this.getMeta("publicKey")].includes(this.recipient());
	}

	public isTransfer(): boolean {
		return this.data.typeGroup === 1 && this.data.type === 0;
	}

	public isSecondSignature(): boolean {
		return this.data.typeGroup === 1 && this.data.type === 1;
	}

	public isDelegateRegistration(): boolean {
		return this.data.typeGroup === 1 && this.data.type === 2;
	}

	public isVote(): boolean {
		const isVote = this.data.typeGroup === 1 && this.data.type === 3;

		if (!isVote) {
			return false;
		}

		return (this.asset().votes as string[])[0].startsWith("+");
	}

	public isUnvote(): boolean {
		const isVote = this.data.typeGroup === 1 && this.data.type === 3;

		if (!isVote) {
			return false;
		}

		return (this.asset().votes as string[])[0].startsWith("-");
	}

	public isMultiSignature(): boolean {
		return this.data.typeGroup === 1 && this.data.type === 4;
	}

	public isIpfs(): boolean {
		return this.data.typeGroup === 1 && this.data.type === 5;
	}

	public isMultiPayment(): boolean {
		return this.data.typeGroup === 1 && this.data.type === 6;
	}

	public isDelegateResignation(): boolean {
		return this.data.typeGroup === 1 && this.data.type === 7;
	}

	public isHtlcLock(): boolean {
		return this.data.typeGroup === 1 && this.data.type === 8;
	}

	public isHtlcClaim(): boolean {
		return this.data.typeGroup === 1 && this.data.type === 9;
	}

	public isHtlcRefund(): boolean {
		return this.data.typeGroup === 1 && this.data.type === 10;
	}

	public isBusinessRegistration(): boolean {
		return this.data.typeGroup === 2 && this.data.type === 0;
	}

	public isBusinessResignation(): boolean {
		return this.data.typeGroup === 2 && this.data.type === 1;
	}

	public isBusinessUpdate(): boolean {
		return this.data.typeGroup === 2 && this.data.type === 2;
	}

	public isBridgechainRegistration(): boolean {
		return this.data.typeGroup === 2 && this.data.type === 3;
	}

	public isBridgechainResignation(): boolean {
		return this.data.typeGroup === 2 && this.data.type === 4;
	}

	public isBridgechainUpdate(): boolean {
		return this.data.typeGroup === 2 && this.data.type === 5;
	}

	public isEntityRegistration(): boolean {
		return this.data.typeGroup === 2 && this.data.type === 6 && this.data.asset.action === 0;
	}

	public isEntityResignation(): boolean {
		return this.data.typeGroup === 2 && this.data.type === 6 && this.data.asset.action === 1;
	}

	public isEntityUpdate(): boolean {
		return this.data.typeGroup === 2 && this.data.type === 6 && this.data.asset.action === 2;
	}
}
