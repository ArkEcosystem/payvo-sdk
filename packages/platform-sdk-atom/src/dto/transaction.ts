import { Contracts, DTO, Exceptions } from "@arkecosystem/platform-sdk";
import { DateTime } from "@arkecosystem/platform-sdk-intl";
import { BigNumber } from "@arkecosystem/platform-sdk-support";

export class TransactionData extends DTO.AbstractTransactionData implements Contracts.TransactionData {
	public id(): string {
		return this.data.txhash;
	}

	public blockId(): string | undefined {
		return undefined;
	}

	public timestamp(): DateTime | undefined {
		return DateTime.make(this.data.timestamp);
	}

	public confirmations(): BigNumber {
		return BigNumber.ZERO;
	}

	public sender(): string {
		const event = this.data.events.find((event) => event.type === "message");
		const attribute = event.attributes.find((attribute) => attribute.key === "sender");

		return attribute.value;
	}

	public recipient(): string {
		const event = this.data.events.find((event) => event.type === "transfer");
		const attribute = event.attributes.find((attribute) => attribute.key === "recipient");

		return attribute.value;
	}

	public recipients(): Contracts.MultiPaymentRecipient[] {
		return [];
	}

	// @ts-ignore
	public amount(): BigNumber {
		const event = this.data.events.find((event) => event.type === "transfer");
		const attribute = event.attributes.find((attribute) => attribute.key === "amount");

		return BigNumber.make(attribute.value.replace(/\D/g, ""));
	}

	public fee(): BigNumber {
		return BigNumber.make(this.data.gas_used);
	}

	public memo(): string | undefined {
		return this.data.tx.value.memo;
	}

	public asset(): Record<string, unknown> {
		return {};
	}

	public isConfirmed(): boolean {
		return false;
	}

	public isSent(): boolean {
		return false;
	}

	public isReceived(): boolean {
		return false;
	}

	public isTransfer(): boolean {
		return false;
	}

	public isSecondSignature(): boolean {
		return false;
	}

	public isDelegateRegistration(): boolean {
		return false;
	}

	public isVote(): boolean {
		return false;
	}

	public isUnvote(): boolean {
		return false;
	}

	public isMultiSignature(): boolean {
		return false;
	}

	public isIpfs(): boolean {
		return false;
	}

	public isMultiPayment(): boolean {
		return false;
	}

	public isDelegateResignation(): boolean {
		return false;
	}

	public isHtlcLock(): boolean {
		return false;
	}

	public isHtlcClaim(): boolean {
		return false;
	}

	public isHtlcRefund(): boolean {
		return false;
	}

	public isEntityRegistration(): boolean {
		return false;
	}

	public isEntityResignation(): boolean {
		return false;
	}

	public isEntityUpdate(): boolean {
		return false;
	}

	public isBusinessEntityRegistration(): boolean {
		return false;
	}

	public isBusinessEntityResignation(): boolean {
		return false;
	}

	public isBusinessEntityUpdate(): boolean {
		return false;
	}

	public isDeveloperEntityRegistration(): boolean {
		return false;
	}

	public isDeveloperEntityResignation(): boolean {
		return false;
	}

	public isDeveloperEntityUpdate(): boolean {
		return false;
	}

	public isCorePluginEntityRegistration(): boolean {
		return false;
	}

	public isCorePluginEntityResignation(): boolean {
		return false;
	}

	public isCorePluginEntityUpdate(): boolean {
		return false;
	}

	public isDesktopPluginEntityRegistration(): boolean {
		return false;
	}

	public isDesktopPluginEntityResignation(): boolean {
		return false;
	}

	public isDesktopPluginEntityUpdate(): boolean {
		return false;
	}

	public isDelegateEntityRegistration(): boolean {
		return false;
	}

	public isDelegateEntityResignation(): boolean {
		return false;
	}

	public isDelegateEntityUpdate(): boolean {
		return false;
	}

	public isLegacyBusinessRegistration(): boolean {
		return false;
	}

	public isLegacyBusinessResignation(): boolean {
		return false;
	}

	public isLegacyBusinessUpdate(): boolean {
		return false;
	}

	public isLegacyBridgechainRegistration(): boolean {
		return false;
	}

	public isLegacyBridgechainResignation(): boolean {
		return false;
	}

	public isLegacyBridgechainUpdate(): boolean {
		return false;
	}
}
