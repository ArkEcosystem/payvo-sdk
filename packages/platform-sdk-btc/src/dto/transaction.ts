import { Contracts, DTO, Exceptions } from "@arkecosystem/platform-sdk";
import { DateTime } from "@arkecosystem/platform-sdk-intl";
import { BigNumber } from "@arkecosystem/platform-sdk-support";

export class TransactionData extends DTO.AbstractTransactionData implements Contracts.TransactionData {
	public id(): string {
		return this.data.txid;
	}

	public blockId(): string | undefined {
		return undefined;
	}

	public timestamp(): DateTime | undefined {
		return DateTime.make(this.data.blockTime);
	}

	public confirmations(): BigNumber {
		return BigNumber.make(this.data.confirmations);
	}

	public sender(): string {
		throw new Exceptions.NotImplemented(this.constructor.name, "sender");
	}

	public recipient(): string {
		throw new Exceptions.NotImplemented(this.constructor.name, "recipient");
	}

	public recipients(): Contracts.MultiPaymentRecipient[] {
		return [];
	}

	public amount(): BigNumber {
		return BigNumber.make(this.data.value);
	}

	public fee(): BigNumber {
		return BigNumber.make(this.data.fee);
	}

	public memo(): string | undefined {
		return undefined;
	}

	public asset(): Record<string, unknown> {
		return {};
	}

	public isConfirmed(): boolean {
		return false;
	}

	public isSent(): boolean {
		return false;
	}

	public isReceived(): boolean {
		return false;
	}

	public isTransfer(): boolean {
		return false;
	}

	public isSecondSignature(): boolean {
		return false;
	}

	public isDelegateRegistration(): boolean {
		return false;
	}

	public isVote(): boolean {
		return false;
	}

	public isUnvote(): boolean {
		return false;
	}

	public isMultiSignature(): boolean {
		return false;
	}

	public isIpfs(): boolean {
		return false;
	}

	public isMultiPayment(): boolean {
		return false;
	}

	public isDelegateResignation(): boolean {
		return false;
	}

	public isHtlcLock(): boolean {
		return false;
	}

	public isHtlcClaim(): boolean {
		return false;
	}

	public isHtlcRefund(): boolean {
		return false;
	}

	public isEntityRegistration(): boolean {
		return false;
	}

	public isEntityResignation(): boolean {
		return false;
	}

	public isEntityUpdate(): boolean {
		return false;
	}

	public isBusinessEntityRegistration(): boolean {
		return false;
	}

	public isBusinessEntityResignation(): boolean {
		return false;
	}

	public isBusinessEntityUpdate(): boolean {
		return false;
	}

	public isDeveloperEntityRegistration(): boolean {
		return false;
	}

	public isDeveloperEntityResignation(): boolean {
		return false;
	}

	public isDeveloperEntityUpdate(): boolean {
		return false;
	}

	public isCorePluginEntityRegistration(): boolean {
		return false;
	}

	public isCorePluginEntityResignation(): boolean {
		return false;
	}

	public isCorePluginEntityUpdate(): boolean {
		return false;
	}

	public isDesktopPluginEntityRegistration(): boolean {
		return false;
	}

	public isDesktopPluginEntityResignation(): boolean {
		return false;
	}

	public isDesktopPluginEntityUpdate(): boolean {
		return false;
	}

	public isDelegateEntityRegistration(): boolean {
		return false;
	}

	public isDelegateEntityResignation(): boolean {
		return false;
	}

	public isDelegateEntityUpdate(): boolean {
		return false;
	}

	public isLegacyBusinessRegistration(): boolean {
		return false;
	}

	public isLegacyBusinessResignation(): boolean {
		return false;
	}

	public isLegacyBusinessUpdate(): boolean {
		return false;
	}

	public isLegacyBridgechainRegistration(): boolean {
		return false;
	}

	public isLegacyBridgechainResignation(): boolean {
		return false;
	}

	public isLegacyBridgechainUpdate(): boolean {
		return false;
	}
}
