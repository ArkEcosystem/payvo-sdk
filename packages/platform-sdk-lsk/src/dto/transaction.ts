import { Contracts, DTO } from "@arkecosystem/platform-sdk";
import { DateTime } from "@arkecosystem/platform-sdk-intl";
import { BigNumber } from "@arkecosystem/platform-sdk-support";

export class TransactionData extends DTO.AbstractTransactionData implements Contracts.TransactionData {
	// readonly #types = {
	// 	0: "transfer",
	// 	1: "secondSignature",
	// 	2: "delegateRegistration",
	// 	3: "vote",
	// 	4: "multiSignature",
	// };

	// // TODO: remove these once live/beta handling is properly implemented
	// readonly #betanetTypes = {
	// 	8: "transfer",
	// 	9: "secondSignature",
	// 	10: "delegateRegistration",
	// 	11: "vote",
	// 	12: "multiSignature",
	// };

	public id(): string {
		return this.data.id;
	}

	public blockId(): string | undefined {
		return this.data.blockId;
	}

	public timestamp(): DateTime | undefined {
		// TODO: use a genesis timestamp that matches the network
		return DateTime.make("2016-05-24T17:00:00.000Z").addSeconds(this.data.timestamp);
	}

	public confirmations(): BigNumber {
		return BigNumber.make(this.data.confirmations);
	}

	public sender(): string {
		return this.data.senderPublicKey;
	}

	public recipient(): string {
		return this.data.recipientId;
	}

	public recipients(): Contracts.MultiPaymentRecipient[] {
		return [];
	}

	public amount(): BigNumber {
		return BigNumber.make(this.data.amount);
	}

	public fee(): BigNumber {
		return BigNumber.make(this.data.fee);
	}

	public memo(): string | undefined {
		return this.data.asset.data;
	}

	public asset(): Record<string, unknown> {
		return {};
	}

	public isConfirmed(): boolean {
		return false;
	}

	public isSent(): boolean {
		return false;
	}

	public isReceived(): boolean {
		return false;
	}

	public isTransfer(): boolean {
		return this.data.type === 8;
	}

	public isSecondSignature(): boolean {
		return this.data.type === 9;
	}

	public isDelegateRegistration(): boolean {
		return this.data.type === 10;
	}

	public isVote(): boolean {
		if (this.data.type !== 11) {
			return false;
		}

		for (const vote of this.asset().votes as string[]) {
			if (vote.startsWith("+")) {
				return true;
			}
		}

		return false;
	}

	public isUnvote(): boolean {
		if (this.data.type !== 11) {
			return false;
		}

		for (const vote of this.asset().votes as string[]) {
			if (vote.startsWith("-")) {
				return true;
			}
		}

		return false;
	}

	public isMultiSignature(): boolean {
		return this.data.type === 12;
	}

	public isIpfs(): boolean {
		return false;
	}

	public isMultiPayment(): boolean {
		return false;
	}

	public isDelegateResignation(): boolean {
		return false;
	}

	public isHtlcLock(): boolean {
		return false;
	}

	public isHtlcClaim(): boolean {
		return false;
	}

	public isHtlcRefund(): boolean {
		return false;
	}

	public isBusinessRegistration(): boolean {
		return false;
	}

	public isBusinessResignation(): boolean {
		return false;
	}

	public isBusinessUpdate(): boolean {
		return false;
	}

	public isBridgechainRegistration(): boolean {
		return false;
	}

	public isBridgechainResignation(): boolean {
		return false;
	}

	public isBridgechainUpdate(): boolean {
		return false;
	}

	public isEntityRegistration(): boolean {
		return false;
	}

	public isEntityResignation(): boolean {
		return false;
	}

	public isEntityUpdate(): boolean {
		return false;
	}
}
