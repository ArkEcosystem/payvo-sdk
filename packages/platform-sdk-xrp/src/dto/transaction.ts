import { Contracts, DTO, Exceptions } from "@arkecosystem/platform-sdk";
import { DateTime } from "@arkecosystem/platform-sdk-intl";
import { BigNumber } from "@arkecosystem/platform-sdk-support";
import BN from "bignumber.js";

export class TransactionData extends DTO.AbstractTransactionData implements Contracts.TransactionData {
	public id(): string {
		return this.data.id;
	}

	public blockId(): string | undefined {
		return undefined;
	}

	public timestamp(): DateTime | undefined {
		if (!this.data.outcome.timestamp) {
			return undefined;
		}

		return DateTime.make(this.data.outcome.timestamp);
	}

	public confirmations(): BigNumber {
		return BigNumber.ZERO;
	}

	public sender(): string {
		return this.data.specification.source.address;
	}

	public recipient(): string {
		return this.data.specification.destination.address;
	}

	public recipients(): Contracts.MultiPaymentRecipient[] {
		return [];
	}

	public amount(): BigNumber {
		const satoshi: string = new BN(this.data.outcome.deliveredAmount.value).times(1e8).toFixed();

		return BigNumber.make(satoshi);
	}

	public fee(): BigNumber {
		const satoshi: string = new BN(this.data.outcome.fee).times(1e8).toFixed();

		return BigNumber.make(satoshi);
	}

	public memo(): string | undefined {
		return undefined;
	}

	public asset(): Record<string, unknown> {
		return {};
	}

	public isConfirmed(): boolean {
		return false;
	}

	public isSent(): boolean {
		return false;
	}

	public isReceived(): boolean {
		return false;
	}

	public isTransfer(): boolean {
		return false;
	}

	public isSecondSignature(): boolean {
		return false;
	}

	public isDelegateRegistration(): boolean {
		return false;
	}

	public isVote(): boolean {
		return false;
	}

	public isUnvote(): boolean {
		return false;
	}

	public isMultiSignature(): boolean {
		return false;
	}

	public isIpfs(): boolean {
		return false;
	}

	public isMultiPayment(): boolean {
		return false;
	}

	public isDelegateResignation(): boolean {
		return false;
	}

	public isHtlcLock(): boolean {
		return false;
	}

	public isHtlcClaim(): boolean {
		return false;
	}

	public isHtlcRefund(): boolean {
		return false;
	}

	public isEntityRegistration(): boolean {
		return false;
	}

	public isEntityResignation(): boolean {
		return false;
	}

	public isEntityUpdate(): boolean {
		return false;
	}

	public isBusinessEntityRegistration(): boolean {
		return false;
	}

	public isBusinessEntityResignation(): boolean {
		return false;
	}

	public isBusinessEntityUpdate(): boolean {
		return false;
	}

	public isDeveloperEntityRegistration(): boolean {
		return false;
	}

	public isDeveloperEntityResignation(): boolean {
		return false;
	}

	public isDeveloperEntityUpdate(): boolean {
		return false;
	}

	public isCorePluginEntityRegistration(): boolean {
		return false;
	}

	public isCorePluginEntityResignation(): boolean {
		return false;
	}

	public isCorePluginEntityUpdate(): boolean {
		return false;
	}

	public isDesktopPluginEntityRegistration(): boolean {
		return false;
	}

	public isDesktopPluginEntityResignation(): boolean {
		return false;
	}

	public isDesktopPluginEntityUpdate(): boolean {
		return false;
	}

	public isDelegateEntityRegistration(): boolean {
		return false;
	}

	public isDelegateEntityResignation(): boolean {
		return false;
	}

	public isDelegateEntityUpdate(): boolean {
		return false;
	}

	public isLegacyBusinessRegistration(): boolean {
		return false;
	}

	public isLegacyBusinessResignation(): boolean {
		return false;
	}

	public isLegacyBusinessUpdate(): boolean {
		return false;
	}

	public isLegacyBridgechainRegistration(): boolean {
		return false;
	}

	public isLegacyBridgechainResignation(): boolean {
		return false;
	}

	public isLegacyBridgechainUpdate(): boolean {
		return false;
	}
}
