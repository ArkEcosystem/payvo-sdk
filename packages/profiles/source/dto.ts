export * from "./signed-transaction.dto.js";
export * from "./transaction.collection";
export * from "./transaction.dto.js";
export * from "./transaction.mapper";
