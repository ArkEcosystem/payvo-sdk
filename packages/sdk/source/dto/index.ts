export * from "./confirmed-transaction.js";
export * from "./signed-transaction.contract.js";
export * from "./signed-transaction.js";
export * from "./unspent-transaction.js";
export * from "./wallet.js";
