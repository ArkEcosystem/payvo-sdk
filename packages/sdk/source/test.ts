/* istanbul ignore file */

import { randomUUID } from "crypto";
import { ConfigKey, ConfigRepository } from "./coins";
import { Container, BindingType } from "./ioc";
import { NetworkManifest } from "./networks";
import { BigNumberService } from "./services";

const createContainer = ({
	config,
	httpClient,
	manifest,
	meta,
}: {
	config?: ConfigRepository;
	httpClient: any;
	manifest: NetworkManifest;
	meta?: any;
}): Container => {
	config ??= new ConfigRepository({ network: manifest.id, httpClient });

	config.set(ConfigKey.Network, manifest);

	if (meta) {
		for (const [key, value] of Object.entries(meta)) {
			config.set(key, value);
		}
	}

	const container = new Container();
	container.constant(BindingType.ConfigRepository, config);
	container.constant(BindingType.HttpClient, httpClient);
	container.singleton(BindingType.BigNumberService, BigNumberService);

	return container;
};

export const createService = <T = any>({
	config,
	httpClient,
	manifest,
	meta,
	predicate,
	service,
}: {
	config?: ConfigRepository;
	httpClient: any;
	manifest: NetworkManifest;
	meta?: any;
	predicate: any;
	service: any;
}): T => {
	const container = createContainer({
		config,
		httpClient,
		manifest,
		meta,
	});

	if (predicate) {
		predicate(container);
	}

	return container.resolve(service);
};

export const createServiceAsync = async <T = any>({
	config,
	httpClient,
	manifest,
	meta,
	predicate,
	service,
}: {
	config?: ConfigRepository;
	httpClient: any;
	manifest: NetworkManifest;
	meta?: any;
	predicate: any;
	service: any;
}): Promise<T> => {
	const container = createContainer({
		config,
		httpClient,
		manifest,
		meta,
	});

	if (predicate) {
		await predicate(container);
	}

	const uuid: string = randomUUID();

	container.singleton(uuid, service);

	return container.get(uuid);
};
