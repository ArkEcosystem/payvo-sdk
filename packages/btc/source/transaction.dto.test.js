import { DateTime } from "@payvo/sdk-intl";

import Fixture from "../test/fixtures/client/transaction.json";
import { createService } from "../test/mocking";
import { ConfirmedTransactionData } from "./confirmed-transaction.dto";

let subject;

test.before.each(async () => {
	subject = await createService(ConfirmedTransactionData);
	subject.configure(Fixture.data);
});

describe("ConfirmedTransactionData", () => {
	test("should succeed", async () => {
		assert.instance(subject, ConfirmedTransactionData);
		assert.is(subject.id(), "21c0cdf1d1e191823540841dd926944e7bc4ee37a7227ec9609ad9715227a02d");
		assert.is(subject.type(), "transfer");
		assert.instance(subject.timestamp(), DateTime);
		assert.is(subject.confirmations().toNumber(), 123456);

		assert.is(subject.sender(), "1Ct7Aivo3jBhabLW8MRkzf28M1QHuqDWCg");
		assert.is(subject.senders()).toMatchSnapshot();

		assert.is(subject.recipient(), "1DVGtxX1ox92cQ5uMrXBL8snE3Agkt9zPr");
		assert.is(subject.recipients()).toMatchSnapshot();

		assert.is(subject.amount().toNumber(), 62550000);
		assert.is(subject.fee().toNumber(), 50000);

		assert.is(subject.inputs()).toMatchSnapshot();
		assert.is(subject.outputs()).toMatchSnapshot();
	});
});
