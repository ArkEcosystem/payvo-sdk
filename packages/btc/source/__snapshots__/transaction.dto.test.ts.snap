// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmedTransactionData should succeed 1`] = `
Array [
  Object {
    "address": "1Ct7Aivo3jBhabLW8MRkzf28M1QHuqDWCg",
    "amount": 62600000,
  },
]
`;

exports[`ConfirmedTransactionData should succeed 2`] = `
Array [
  Object {
    "address": "1DVGtxX1ox92cQ5uMrXBL8snE3Agkt9zPr",
    "amount": 62050000,
  },
  Object {
    "address": "19N3CU2B775pk9E38Ebs3EwhAyrPUBZ7Jz",
    "amount": 100000,
  },
  Object {
    "address": "1FaT9yh2UVJW68ugwvoMkmZPemZjpduX9B",
    "amount": 100000,
  },
  Object {
    "address": "12C1rVsgUUNKfFYWQ9X18M38c4hsGV9T5w",
    "amount": 100000,
  },
  Object {
    "address": "1NfjJVvACPArTywVPraMQPyie5mL9NkHbF",
    "amount": 100000,
  },
  Object {
    "address": "13RCWHB1btpym5vX7Yu9Gq9ki66VVvLABy",
    "amount": 100000,
  },
]
`;

exports[`ConfirmedTransactionData should succeed 3`] = `
Array [
  UnspentTransactionData {},
]
`;

exports[`ConfirmedTransactionData should succeed 4`] = `
Array [
  UnspentTransactionData {},
  UnspentTransactionData {},
  UnspentTransactionData {},
  UnspentTransactionData {},
  UnspentTransactionData {},
  UnspentTransactionData {},
]
`;
